client/containers/Pub/PubSuspendWhileTyping.tsx

```diff
@@ -49,30 +47,14 @@ export const PubSuspendWhileTypingProvider = (props: PubSuspendWhileTypingProvid
 	);
 };
 
-type MemoizedSuspendProps = {
-	shouldUpdate: boolean;
-	children: ChildrenFn;
-};
-
-const MemoizedSuspend = React.memo<MemoizedSuspendProps>(
-	// eslint-disable-next-line react/prop-types
-	(props) => props.children(),
-	// Return false (indicating that the memoized version is invalid) if we should update
-	(_, nextProps) => !nextProps.shouldUpdate,
-);
-
-type PubSuspendWhileTypingProps = {
-	children: ChildrenFn;
-	delay: number;
-};
-
-export const PubSuspendWhileTyping = (props: PubSuspendWhileTypingProps) => {
-	const { children, delay } = props;
+export const usePubSuspendWhileTyping = <T extends any>(value: T, delay = 1000): T => {
```

Comment:
Why use <T extends any> instead of <T extends PubContextType>?

Reply:
I'd say it's for separation of concerns: this hook works fine regardless of what kind of value you pass it, so there's no reason to more tightly bind it to `usePubContext()`.

The reason I wrote `<T extends any>` instead of `<T>` is much sillier. In a .tsx file, I guess the parser can't tell the difference between `<T>` the generic parameter and `<T>` as an opening JSX tag, so explicitly writing `<T extends any>` is just to help it out!